pytorch_lightning/utilities/distributed.py

```diff
@@ -44,8 +44,18 @@ def wrapped_fn(*args, **kwargs):
     return wrapped_fn
 
 
+# TODO: this should be part of the cluster environment
+def _get_local_rank() -> int:
+    local_rank_keys = ('LOCAL_RANK', 'SLURM_LOCALID')
```

Comment:
an nasty way :D
`os.environ.get('LOCAL_RANK', os.environ.get('SLURM_LOCALID', 0))`

Reply:
in case we have another cluster to support and we haven't fixed the design issues here, the tuple + loop will be easier to extend. hopefully it doesn't come to that though and we can remove this altogether.

Reply:
Should be SLURM_PROCID to get the global rank